src/FSharpVSPowerTools.Core/AssemblyContentProvider.fs

```diff
@@ -184,29 +184,33 @@ module AssemblyContentProvider =
             |> Seq.map (traverseEntity contentType Parent.Empty)
             |> Seq.concat
             |> Seq.distinct
-            |> Seq.toList
 
     let private getAssemblySignaturesContent contentType (assemblies: FSharpAssembly list) = 
         assemblies 
-        |> List.map (fun asm -> getAssemblySignatureContent contentType asm.Contents)
-        |> List.concat 
+        |> Seq.map (fun asm -> getAssemblySignatureContent contentType asm.Contents)
+        |> Seq.concat
+        |> Seq.toList
+
+    let private entityCache = Dictionary<AssemblyPath, DateTime * AssemblyContentType * RawEntity list>()
```

Comment:
I wonder whether we could move this cache and `getAssemblyContent` function to `Logic` project? Having a cache in the core library seems like a leak of abstraction to me. If it is in the `Logic` project, we could invalidate the cache (e.g. after closing solutions) if it's necessary to do so.